docs/naps/3-spaces.md

```diff
@@ -0,0 +1,127 @@
+
+Currently, to do so, a user is forced to either:
+- load everything into the layer list (thus "pretending" that the absolute coordinates do indeed match) and then develop a plugin or widget that manages layer visibility
+- or develop a plugin or widget that manages layers externally and feeds them to `Viewer.layers` as desired
+- do nothing and deal with hundreds of layers manually
+
+While this might not seem problematic with few images, it quickly degenerates when working with big datasets, especially in 3D (for an example use case, check out [this issue comment](https://github.com/napari/napari/issues/4419#issuecomment-1113090992)).
+
+The above workarounds have the following issues:
+- *usability*: adding too many layers to the viewer makes navigating the layerlist unwieldy, a problem that cannot be solved by layer-groups when there is no reasonable way to group layers
+- *usability*: forcing the creation of a plugin or widget with custom logic is one more barrier for non-developers, and possibly one more meta-object (e.g: a "dataset") that the user has to juggle around and that napari is unaware of.
+- *performance*: many layers perform worse than few layers. This might be addressed separately, but is currently unresolved.
+- *abstraction*: it makes no sense for things to share coordinate space when they shouldn't.
+- *performance and usability*: regardless of the workaround used, reader plugins can't know about it, so a user would be forced to either first load everything into the viewer and then do something with it, significantly slowing down startup and adding one more step to opening napari
+- *serialization*: without a native `napari` support, these workarounds cannot be properly serialized.
+
+Additionally, while this was not the main goal of this NAP, people have sometimes asked for "workspaces", where different workflows can be tested in parallel; `spaces` would also implicitly allow this by letting users re-use a layer in multiple spaces.
+
+# Implementation Proposal
+
+## API
+
+`ViewerModel.spaces` would be a (selectable) `EventedList` or similar evented collection, each containing a `Space` object, with the following attributes:
+- `name`: unique name for the space
+- `layers`: a `layerlist` (or top-level `layergroup`, in the future)
+- `active`: similar to how a layer in a `layerlist` can be active, a `Space` in the `spaces` can be active; this will determine which space is loaded into the layerlist and used to populate the canvas
+- `camera`: a snapshot of the `Camera` model
+- `dims`: a snapshot of the `Dims` model
+- ...
+
+In general, the most intuitive approach would be to mirror the basic `Layer` api (i.e: with `name`, `metadata`, possibly `source`...).
```

Comment:
`metadata` is always useful; if someone wants to mark each space with (for example) different experimental conditions, that should be possible.

`source` could be populated the same way as layers, in case (as mentioned in the `Plugins` section) a reader were to return spaces rather than just layers (if that part of the proposal gets accepted).